Update positron cli and linux package info with positron version output

sharon-wang opened the pull request:
## Description

- addresses #1373
- updates `positron --version`, `positron --help`, `positron-server --version`, `positron-server --help` to output Positron version information
- updates Linux package version info to use Positron version
- I updated our snapcraft.yml, but we don't publish on Snap yet, so it's just a speculative change for now

## QA Notes

Please test this functionality in each of our builds across all platforms! I've tested this with Desktop Mac and REH-Web on Ubuntu.

Note that the Command Prompt command to install `positron` in PATH is not available on every operating system. On some platforms, it seems that `positron` is automatically installed in the PATH upon installation.

### Check `--version`

Sample Output:
```
Positron: 2024.11.0 build 70
Positron SHA: 51829054cdff5f1c0464f27fb223921cf9c7f427
Code OSS: 1.93.0
Arch: arm64
```

#### Desktop Build
1. Use the Command Prompt to `Shell Command: Install 'positron' command in PATH` (`workbench.action.installCommandLine`)
2. Run `positron --version` in a Terminal
3. Confirm the output contains the expected information and format

#### REH or REH Web Build
1. Locate the `positron-server` executable, which will be in the `bin` directory of the REH or REH Web binary
2. Run `positron-server --version` in a Terminal
3. Confirm the output contains the expected information and format

### Check `--help`

Sample Output:
```
Positron 2024.11.0
<REST_OF_HELP>
```

#### Desktop Build
1. Use the Command Prompt to `Shell Command: Install 'positron' command in PATH` (`workbench.action.installCommandLine`)
2. Run `positron --help` in a Terminal
3. The first line of output should show `Positron <POSITRON_VERSION>`

#### REH or REH Web Build
1. Locate the `positron-server` executable, which will be in the `bin` directory of REH or REH Web binary
2. Run `positron-server --help` in a Terminal
3. The first line of output should show `Positron <POSITRON_VERSION>`

### Check package info

See https://github.com/posit-dev/positron/pull/5140 for notes on how to verify the version output of DEB and RPM packages.

sharon-wang committed:
update cli version info to use `positronVersion` instead of `version`

sharon-wang committed:
pass `positronVersion` where `@@VERSION@@` is used

sharon-wang committed:
update linux release version info

sharon-wang committed:
add positron version and build number to CLI version output

sharon-wang committed:
update gulpfile.vscode.js with positron version and build number

although `resources/win32/bin/code.sh` doesn't actually use `@@VERSION@@`, `@@POSITRONVERSION@@`, or `@@BUILDNUMBER@@`, I've updated the code in case it's needed in the future.

sharon-wang committed:
update format of version string

sharon-wang committed:
nit for variable case

sharon-wang committed:
update help message with positron version

sharon-wang committed:
add missing code fences

sharon-wang commented:
Once this is looking good from a review perspective, I'll run a full build for all platforms and work with QA to do a round of verification before merging.

petetronic commented:
Built a version on Mac OS locally and confirmed the cli tool binary inside the .app generated reported:
```
Positron: 2024.11.0 build 77
Positron SHA: 668f906f2ea7b3100f3b0b3ab57f77b21b7c51fc
Code OSS: 1.93.0
Arch: arm64
```

jmcphers commented on src/vs/platform/environment/node/argv.ts:
```diff
@@ -466,6 +466,31 @@ export function buildHelpMessage(productName: string, executableName: string, ve
 	return help.join('\n');
 }
 
-export function buildVersionMessage(version: string | undefined, commit: string | undefined): string {
```
Nit: For ease of merging, I think it'd be better for us to have a `buildPositronVersionMessage` of our own and leave `buildVersionMesage` alone.

jmcphers reviewed (dismissed):
Didn't test functionality since @petetronic did that, but code LGTM.

sharon-wang committed:
keep but deprecate `buildVersionMessage` in favour of `buildPositronVersionMessage`

sharon-wang committed:
fix linux debian package version format

The deb-version has a specific format, one aspect of which is no spaces.

```
Error: Command failed: fakeroot dpkg-deb -b positron-amd64 deb
dpkg-deb: error: parsing file 'positron-amd64/DEBIAN/control' near line 2 package 'positron':
 'Version' field value '2024.11.0-1729291046 build 0': version string has embedded spaces
```

https://linux.die.net/man/5/deb-version

So, the version we'll include in the package info will be `<version>+<build>-<revision>`

sharon-wang committed:
also remove spaces from RPM spec file

I'm not seeing equivalent docs for RPM that note no spaces in the version string of the spec file, but it probably doesn't hurt to keep the spec template similar to the DEB control template.

sharon-wang committed:
code fence `buildPositronVersionMessage` imports

sharon-wang commented:
Version info is looking good for Linux and Mac Desktop, REH and REH-Web in release builds. There's an issue with the build number not set properly in the DEB and RPM files, but I will address that separately. I also discovered #5101 which can be addressed later/separately.

@jonvanausdeln if Windows is looking good, or if you're comfortable with verifying that after this is in, then I'll merge this!

jonvanausdeln commented:
Windows works as well :)

```
C:\Users\JonVanausdeln> positron -v
Positron: 2024.11.0 build 81
Positron SHA: 033600190d4eef8c18bc766303750dff61c39942
Code OSS: 1.93.0
Arch: x64
```
